### PR TITLE
add utiltiles of network filter

### PR DESCRIPTION
**Pull request by juteman**
the test about network filter has utiltiles code
add in this file

Signed-off-by: juteman <qiulzhan@redhat.com>

### CODE REVIEW

File: virttest/utils_test/utils_nwfilter.py [hunk 1]
```diff
@@ -1,10 +1,79 @@
...
+
+    return:  new network interface
+    """
+    # set iface type name as network and binding source type
+    New_iface.type_name = type_name
+    New_iface.source = source
+    # if filterref_dict is not blank , bind it to iface
+    if filterref_dict:
+        filterref = New_iface.new_filterref(**filterref_dict)
+        New_iface.filterref = filterref
```
Comment (hahakiki2010) [inline virttest/utils_test/utils_nwfilter.py]: sometimes this line doesn't work,  you can try below code. The New_iface is got from existing vmxml and does not ever contain a filterref property. Then accessor check will be failed. 
```
         vmxml = vm_xml.VMXML.new_from_dumpxml(vm_name)
-        xml_devices = vmxml.devices
-        iface_index = xml_devices.index(
-            xml_devices.by_device_tag("interface")[0])
-        iface = xml_devices[iface_index]
-        iface.type_name = "bridge"
-        iface.source = {'bridge': br_name}
-        iface.filterref = iface.new_filterref(name=nwfilter)
-        iface.xmltreefile.write()
```
```
File "/var/tmp/libvirt-ci/runtest/avocado-vt/avocado-vt/virttest/libvirt_xml/accessors.py", line 29, in type_check
    % (name, expected, is_a_name))

ValueError: Instance of Filterref value is not any of [<class 'interface.Filterref'>], it is a <class 'interface.Filterref'>
```

File: virttest/utils_test/utils_nwfilter.py [hunk 2]
```diff
@@ -1,10 +1,79 @@
...
 from virttest.libvirt_xml import base, accessors
 from virttest.libvirt_xml.devices.interface import Filterref
+from virttest import libvirt_xml
+
+# network filter xml set up code BEGIN
+
+
+def set_nwfilter_iface(New_iface,
+                       type_name="network",
+                       source={'network': "default"},
```
Comment (hahakiki2010) [inline virttest/utils_test/utils_nwfilter.py]: do not use default network, as it's reserved by libvirt and normally not allowed to be modified by others. 
I think source={} is fine, and add empty check in code if {} is not allowed

File: virttest/utils_test/utils_nwfilter.py [hunk 3]
```diff
@@ -1,10 +1,79 @@
...
+    # print the new iface xml
+    logging.debug("new iface xml is: \n %s \n" % New_iface)
+    return New_iface
+
+
+def set_filterref_dict(filter_name, params, param_name, param_value):
+    """
+    set up the filter rule dict
+
+    Params filter_name: string, network filter rule name.
```
Comment (hahakiki2010) [inline virttest/utils_test/utils_nwfilter.py]: ```params``` but not Params

File: virttest/utils_test/utils_nwfilter.py [hunk 4]
```diff
@@ -1,10 +1,79 @@
...
+import logging
+
 from virttest.libvirt_xml import base, accessors
 from virttest.libvirt_xml.devices.interface import Filterref
+from virttest import libvirt_xml
+
+# network filter xml set up code BEGIN
+
+
+def set_nwfilter_iface(New_iface,
```
[the remark this answers is not in the document]
Comment (juteman) [reply virttest/utils_test/utils_nwfilter.py]: ok, i will change it .

**Comment by sathnaga** [comment]
@juteman update pls..

**Comment by dzhengfy** [comment]
As no usage for this PR for so long and the author does not work on this any more, we decide to close it. In case of new requirement for this feature, we will file a new PR.

**Pull request closed without merging**